Assets/MixedRealityToolkit.Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs

```diff
@@ -423,27 +554,86 @@ private static void OptimizeRenderer(Renderer renderer)
             renderer.allowOcclusionWhenDynamic = false;
         }
 
-        private static void MemoryUsageToString(StringBuilder stringBuilder, string displayedDecimalFormat, TextMesh textMesh, string memoryString, ulong memoryUsage)
+        private static void MemoryUsageToString(char[] stringBuffer, int displayedDecimalDigits, TextMesh textMesh, string prefixString, ulong memoryUsage)
```

Comment:
My first reaction without context is "Why don't you use `string.Format()`?". Should we add a comment to explain the rationale of manual formatting?

Reply:
Good idea, added!